### PR TITLE
add user permission api, docs and tests and proper permissions

### LINKED ISSUE #3968: Add user_permission api along with docs and testing

### PR DESCRIPTION
**Pull request by shubham-padia**
fixes #3968 and #4070 

Changes:
- Removes `verified_user` field from user_permissions table
- add relevant permission checks in decorators
- Add missing event post test to dredd hooks

### CODE REVIEW

**Comment by shubham-padia** [comment]
Please review

**Comment by shubham-padia** [comment]
@fossasia/open-event-server-devs Please review

File: tests/hook_main.py [hunk 1]
```diff
@@ -2083,3 +2084,66 @@ def custom_placeholder_delete(transaction):
         custom_placeholder = CustomPlaceholderFactory()
         db.session.add(custom_placeholder)
         db.session.commit()
+
+
+# ------------------------- Pages -------------------------
```
Comment (enigmaeth) [inline tests/hook_main.py]: Pages -> User Permissions

File: app/api/user_permission.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    """
+    decorators = (api.has_permission('is_admin', methods="POST"),)
+    schema = UserPermissionSchema
+    data_layer = {'session': db.session,
+                  'model': UserPermission}
+
+
+class UserPermissionDetail(ResourceDetail):
+    """
+    Page detail by id
```
Comment (enigmaeth) [inline app/api/user_permission.py]: Page -> User Permissions

File: tests/hook_main.py (hunk 1 above)
Comment (shubham-padia) [inline tests/hook_main.py]: Resolved

File: app/api/user_permission.py (hunk 2 above)
Comment (shubham-padia) [inline app/api/user_permission.py]: Resolved

**Pull request merged**